src/Compilers/CSharp/Portable/BoundTree/Expression.cs

```diff
@@ -122,17 +157,22 @@ internal static ImmutableArray<IArgument> DeriveArguments(ImmutableArray<BoundEx
                     argumentIndex = argumentsToParameters.IndexOf(parameterIndex);
                 }
 
-                if (argumentIndex == -1)
+                // No argument has been supplied for the parameter at `parameterIndex`:
+                // 1. `argumentIndex == -1' when when the arguments are specified out of parameter order, and no argument is provided for parameter corresponding to `parameters[parameterIndex]`.
+                // 2. `argumentIndex >= boundArguments.Length` when the arguments are specified in parameter order, and no argument is provided at `parameterIndex`.
+                if (argumentIndex == -1 || argumentIndex >= boundArguments.Length)
                 {
-                    // No argument has been supplied for the parameter.
                     Symbols.ParameterSymbol parameter = parameters[parameterIndex];
+                    // Corresponding parameter is optional with default value.
                     if (parameter.HasExplicitDefaultValue)
                     {
                         arguments.Add(new Argument(ArgumentKind.DefaultValue, parameter, new Literal(parameter.ExplicitDefaultConstantValue, parameter.Type, null)));
```

Comment:
It doesn't seem to be the case. I will create a bug for this as it is unrelated to this PR.